src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -153,18 +153,18 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
 
         $url = '';
         $optional = true;
+        $message = 'Parameter "{parameter}" for route "{route}" must match "{expected}" ("{given}" given) to generate a corresponding URL.';
```

Comment:
this string is created even if not used on each route. this should not be happening.